server/src/main/java/io/druid/server/http/DatasourcesResource.java

```diff
@@ -551,8 +551,10 @@ public int compare(DruidDataSource druidDataSource, DruidDataSource druidDataSou
       }
 
       // segment stats
-      totalSegmentCount += druidDataSource.getSegments().size();
-      totalSegmentSize += dataSourceSegmentSize;
+      if (dataSourceSegmentSize >= totalSegmentSize) {
```

Comment:
Been digging through the code and I can't seem to find a place where we maintain a map of datasource names to a set of segments ids. 

So, I think the proper way to fix the issue to create a set with all segments ids associated with the datasource name and figure out the total number of segments and segment sizes from that data structure.